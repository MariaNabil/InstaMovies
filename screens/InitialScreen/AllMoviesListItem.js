import React, {PureComponent} from 'react';
import {Image, Text, View, TouchableOpacity} from 'react-native';
import Constants from '../../utils/Constants';
import {TrimText} from '../../utils/Text';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {shadowStyle3, shadowStyle0, shadowStyle2} from '../../utils/Styles';
import FastImage from 'react-native-fast-image';

export default class AllMoviesListItem extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {};
  }

  renderTitle = (title) => {
    if (title) {
      return (
        <Text
          style={{
            fontWeight: 'bold',
            fontSize: 17,
            paddingBottom: 5,
            width: '70%',
          }}
          numberOfLines={2}>
          {title}
        </Text>
      );
    } else return null;
  };

  renderOverview = (overview) => {
    if (overview) {
      return (
        <View style={{flexDirection: 'row'}}>
          <Text style={{paddingBottom: 5}}>{TrimText(overview, 100)}</Text>
        </View>
      );
    } else {
      return null;
    }
  };

  renderDate = (date) => {
    if (date) {
      return (
        <View
          style={{
            alignItems: 'flex-end',
            justifyContent: 'flex-end',
            alignSelf: 'flex-end',
          }}>
          <Text style={{textAlign: 'right'}}>{date}</Text>
        </View>
      );
    } else {
      return null;
    }
  };

  renderPoster = (poster_path) => {
    let isPlaceholder = false;
    if (!poster_path || poster_path == '') {
      isPlaceholder = true;
    }
    let uri = `${Constants.IMAGE_BASE_URL}${poster_path}`;
    return (
      <FastImage
        source={
          isPlaceholder
            ? require('../../assets/Images/MoviePlaceholder.jpg')
            : {uri: uri, priority: FastImage.priority.high}
        }
        style={{
          width: '30%',
          height: 150,
          borderRadius: 5,
          borderWidth: 0.3,
          borderColor: 'black',
        }}></FastImage>
    );
  };

  renderExpandButton = () => {
    return (
      <TouchableOpacity
        style={{paddingBottom: 20, paddingLeft: 20}}
        onPress={() => {
          this.props.onPress(this.props.index);
        }}>
        <MaterialIcons name="unfold-more" size={20} />
      </TouchableOpacity>
    );
  };

  render() {
    const {item, index} = this.props;
    const {title, poster_path, overview, release_date} = item;
    return (
      <TouchableOpacity
        style={{
          borderRadius: 5,
          flexDirection: 'row',
          flex: 1,
          // width: '95%',
          padding: 10,
          marginVertical: 5,
          marginHorizontal: 10,
          // justifyContent: 'space-between',
          backgroundColor: Constants.BACKGROUND_COLOR,
          alignSelf: 'center',
          ...shadowStyle2,
        }}
        onPress={() => this.props.onPress(index)}>
        {this.renderPoster(poster_path)}
        <View
          style={{
            paddingLeft: 10,
            width: '70%',
            justifyContent: 'space-between',
          }}>
          <View>
            <View
              style={{flexDirection: 'row', justifyContent: 'space-between'}}>
              {this.renderTitle(title)}
              {this.renderExpandButton()}
            </View>
            {this.renderOverview(overview)}
          </View>
          {this.renderDate(release_date)}
        </View>
      </TouchableOpacity>
    );
  }
}
