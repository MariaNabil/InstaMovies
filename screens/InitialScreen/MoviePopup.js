import React, {PureComponent} from 'react';
import {
  Image,
  Text,
  View,
  TouchableOpacity,
  ScrollView,
  Dimensions,
  SafeAreaView,
} from 'react-native';
import Constants from '../../utils/Constants';
import {shadowStyle3} from '../../utils/Styles';
import Ionicons from 'react-native-vector-icons/Ionicons';
import Modal from 'react-native-modal';

export default class MoviePopup extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {screenWidth: Dimensions.get('window').width};
  }

  componentDidMount() {
    //re render when change orientation
    Dimensions.addEventListener('change', () => {
      this.setState({
        screenWidth: Dimensions.get('window').width,
        screenHeight: Dimensions.get('window').height,
      });
    });
  }

  renderPopupPoster = (poster_path) => {
    let uri = this.props.isMyMoviePressed
      ? poster_path
      : `${Constants.IMAGE_BASE_URL}${poster_path}`;

    let imageHeight;
    if (poster_path) {
      Image.getSize(uri, (width, height) => {
        imageHeight =
          ((this.state.screenWidth * 70) / 100 - 20) * (height / width);

        this.setState({imageHeight: imageHeight});
      });
    } else {
      this.setState({imageHeight: (this.state.screenWidth * 70) / 100 - 20});
    }

    let isPlaceholder = false;
    if (!poster_path || poster_path == '') {
      isPlaceholder = true;
    }
    return (
      <View
        style={{
          width: '70%',
          borderRadius: 5,
          alignItems: 'center',
          justifyContent: 'center',
          marginBottom: 10,
          height: this.state.imageHeight,
          ...shadowStyle3,
        }}>
        <Image
          source={
            isPlaceholder
              ? require('../../assets/Images/MoviePlaceholder.jpg')
              : {uri: uri}
          }
          style={{
            width: '100%',
            height: '100%',
            borderRadius: 5,

            // height: this.state.imageHeight,
            // borderRadius: 5,
            // marginBottom: 10,
          }}></Image>
      </View>
    );
  };

  renderPopupTitle = (title) => {
    if (title && title != '') {
      return (
        <Text
          style={{
            fontWeight: 'bold',
            fontSize: 20,
            paddingVertical: 10,
            width: '80%',
            textAlign: 'center',
            alignSelf: 'center',
          }}>
          {title}
        </Text>
      );
    }
  };

  renderPopupOverview = (overview) => {
    if (overview && overview != '') {
      return (
        <View style={{flexDirection: 'row'}}>
          <Text
            style={{
              paddingBottom: 10,
              width: '80%',
              textAlign: 'center',
              fontSize: 15,
            }}>
            {overview}
          </Text>
        </View>
      );
    }
  };

  renderPopupDate = (date) => {
    if (date && date != '')
      return (
        <View
          style={{
            alignItems: 'flex-end',
            justifyContent: 'flex-end',
            alignSelf: 'flex-end',
            padding: 10,
            paddingTop: 0,
          }}>
          <Text style={{textAlign: 'right'}}>{date}</Text>
        </View>
      );
  };

  renderPopupCloseButton = () => {
    return (
      <TouchableOpacity
        onPress={this.props.onClosePopup}
        style={{
          right: 10,
          top: 10,
          position: 'absolute',
          zIndex: 2,
          ...shadowStyle3,
        }}>
        <Ionicons name="close-circle-outline" color="#444444" size={25} />
      </TouchableOpacity>
    );
  };

  render() {
    const {
      allMoviesData,
      pressedMovieIndex,
      showPopup,
      onClosePopup,
    } = this.props;

    let item = allMoviesData[pressedMovieIndex];
    const {title, poster_path, overview, release_date, image_uri} = item;

    return (
      <Modal
        isVisible={showPopup}
        onBackdropPress={onClosePopup}
        onSwipeComplete={onClosePopup}>
        <View
          style={{
            width: '100%',
            alignSelf: 'center',
            backgroundColor: '#FFF',
            overflow: 'hidden',
            alignItems: 'center',
            justifyContent: 'center',
            borderRadius: 10,
          }}>
          <SafeAreaView>
            <ScrollView
              contentContainerStyle={{
                width: '100%',
                alignSelf: 'center',
                alignItems: 'center',
                justifyContent: 'center',
              }}
              style={{width: '100%'}}>
              {this.renderPopupCloseButton()}

              <View style={{flexDirection: 'row'}}>
                {this.renderPopupTitle(title)}
              </View>
              {this.renderPopupPoster(
                this.props.isMyMoviePressed ? image_uri : poster_path,
              )}

              {this.renderPopupOverview(overview)}
              {this.renderPopupDate(release_date)}
            </ScrollView>
          </SafeAreaView>
        </View>
      </Modal>
    );
  }
}
