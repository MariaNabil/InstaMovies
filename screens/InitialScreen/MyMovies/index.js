import React, {Component} from 'react';
import {
  StyleSheet,
  View,
  FlatList,
  Dimensions,
  Text,
  TouchableOpacity,
} from 'react-native';

import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import LinearGradient from 'react-native-linear-gradient';
import {shadowStyle2} from '../../../utils/Styles';
import MyMoviesListItem from './MyMoviesListItem';
import Constants from '../../../utils/Constants';

export default class MyMovies extends Component {
  constructor(props) {
    super(props);
    this.state = {
      screenWidth: Dimensions.get('window').width,
      currentScroll: 0,
      isScrollButtonShown: false,
      MyMoviesListHeight: 0,
    };
  }

  //re render when change orientation
  componentDidMount() {
    Dimensions.addEventListener('change', () => {
      this.setState({
        screenWidth: Dimensions.get('window').width,
        screenHeight: Dimensions.get('window').height,
      });
    });
  }

  //ReRender My Movies List When Adding New Movie
  onReRender = () => {
    this.setState({});
  };

  //Go To Add Movie Screen
  goToAddMovie = () => {
    this.props.navigation.navigate('Add Movie', {
      onSave: this.onReRender,
    });
  };

  //Set currentScroll Horizontal Position
  handleScroll = (event) => {
    this.setState({currentScroll: event.nativeEvent.contentOffset.x});
  };

  //Scroll To The Next Movie When Pressing On Scroll Button
  incrementScroll = (increment_by_value) => {
    this.setState({
      currentScroll: this.state.currentScroll + increment_by_value,
    });
    this.listRef.scrollToOffset({offset: +increment_by_value});
  };

  onPressScroll = () => {
    this.incrementScroll(this.state.screenWidth / 2);
  };

  renderMyMoviesListItem = ({item, index}) => {
    return (
      <MyMoviesListItem
        item={item}
        index={index}
        onPress={this.props.onPressMyMovie}
      />
    );
  };

  renderMyMoviesFooter = () => {
    return (
      <View>
        {global.MyMovies?.length == 1 ? (
          <TouchableOpacity
            style={[
              styles.myMoviesFooterStyle,
              {
                width: this.state.screenWidth / 2 - 15,
                height: (this.state.screenWidth / 2 - 15) * 1.5 + 30,
              },
            ]}
            onPress={this.goToAddMovie}>
            <TouchableOpacity
              onPress={this.goToAddMovie}
              style={styles.myMoviesFooterCircleStyle}>
              <Ionicons name={`ios-add`} size={45} />
            </TouchableOpacity>
          </TouchableOpacity>
        ) : null}
      </View>
    );
  };

  renderEmptyList = () => {
    return (
      <TouchableOpacity
        style={[
          {
            width: this.state.screenWidth - 20,
          },
          styles.emptyListStyle,
        ]}
        onPress={this.goToAddMovie}>
        <View style={styles.emptyListViewStyle}>
          <Text style={styles.addMoviesButtonTextStyle}>{'Add Movies'}</Text>
        </View>
      </TouchableOpacity>
    );
  };

  renderScrollButton = () => {
    if (global.MyMovies?.length > 2 && this.state.currentScroll == 0) {
      return (
        <TouchableOpacity
          onPress={this.onPressScroll}
          style={[
            {
              height: this.state.MyMoviesListHeight,
            },
            styles.scrollButtonStyle,
          ]}>
          <LinearGradient
            start={{x: 0.0, y: 0.0}}
            end={{x: 1.0, y: 0.0}}
            colors={[
              `rgba(0, 0, 0, 0.02)`,
              `rgba(0, 0, 0, 0.2)`,
              `rgba(0, 0, 0, 0.5)`,
            ]}
            style={[
              {
                height: this.state.MyMoviesListHeight,
              },
              styles.scrollButtonGradientStyle,
            ]}>
            <MaterialIcons
              name={`arrow-forward-ios`}
              size={22}
              color={'white'}
            />
          </LinearGradient>
        </TouchableOpacity>
      );
    }
  };

  renderMyMoviesList = () => {
    return (
      <FlatList
        onLayout={(e) => {
          this.setState({
            MyMoviesListHeight: e.nativeEvent.layout.height,
            MyMoviesListWidth: e.nativeEvent.layout.width,
          });
        }}
        ListFooterComponent={this.renderMyMoviesFooter}
        horizontal={true}
        style={styles.myMoviesListStyle}
        data={global.MyMovies}
        keyExtractor={({id}) => `${id}`}
        renderItem={({item, index}) =>
          this.renderMyMoviesListItem({item, index})
        }
        ListEmptyComponent={this.renderEmptyList}
        ref={(ref) => (this.listRef = ref)}
        onScroll={this.handleScroll}></FlatList>
    );
  };

  renderMyMoviesHeader = () => {
    return (
      <View style={styles.myMoviesHeaderStyle}>
        <Text style={styles.myMoviesHeaderTextStyle}>{'My Movies'}</Text>
      </View>
    );
  };

  renderAddIcon = () => {
    return (
      <TouchableOpacity
        style={styles.addIconStyle}
        onPress={() => {
          this.props.navigation.navigate('Add Movie', {
            onSave: this.onReRender,
          });
        }}>
        <Ionicons name="add-circle-outline" color="#444444" size={25} />
      </TouchableOpacity>
    );
  };

  render() {
    return (
      <View>
        <View style={styles.containerStyle}>
          {this.renderMyMoviesHeader('My Movies')}
          {global.MyMovies &&
          global.MyMovies.length != 0 &&
          global.MyMovies.length != 1
            ? this.renderAddIcon()
            : null}
        </View>
        <View>
          {this.renderMyMoviesList()}
          {this.renderScrollButton()}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  myMoviesFooterStyle: {
    ...shadowStyle2,
    borderRadius: 5,
    backgroundColor: Constants.BACKGROUND_COLOR,
    margin: 5,
    marginTop: 0,
    alignItems: 'center',
    justifyContent: 'center',
  },
  myMoviesFooterCircleStyle: {
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    width: 110,
    height: 110,
    borderRadius: 110 / 2,
    backgroundColor: 'rgba(72, 74, 84, 0.1)',
  },
  emptyListStyle: {
    flex: 1,
    alignItems: 'stretch',
    justifyContent: 'center',
    alignSelf: 'center',
    marginTop: 10,
    marginHorizontal: 5,
  },
  emptyListViewStyle: {
    borderColor: 'black',
    borderWidth: 1,
    borderRadius: 15,
    alignItems: 'center',
    justifyContent: 'center',
  },
  addMoviesButtonTextStyle: {paddingVertical: 5, textAlign: 'center'},
  scrollButtonStyle: {
    position: 'absolute',
    right: 0,
    backgroundColor: 'transparent',
    top: 0,
    width: 30,
  },
  scrollButtonGradientStyle: {
    width: 30,
    justifyContent: 'center',
    alignItems: 'center',
    borderTopLeftRadius: 20,
    borderBottomLeftRadius: 20,
  },
  myMoviesListStyle: {marginHorizontal: 5, marginBottom: 10, marginTop: 5},
  myMoviesHeaderStyle: {marginHorizontal: 10, alignSelf: 'stretch', flex: 1},
  myMoviesHeaderTextStyle: {paddingTop: 10, fontSize: 20, fontWeight: 'bold'},
  addIconStyle: {paddingHorizontal: 10, paddingTop: 10},
  containerStyle: {flexDirection: 'row', justifyContent: 'space-between'},
});
