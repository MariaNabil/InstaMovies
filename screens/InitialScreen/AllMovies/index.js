import React, {Component} from 'react';
import {
  StyleSheet,
  View,
  FlatList,
  Text,
  ActivityIndicator,
} from 'react-native';

import Constants from '../../../utils/Constants';
import AllMoviesListItem from './AllMoviesListItem';

export default class MyMovies extends Component {
  renderFooterIndicator = () => {
    if (this.props.loading) {
      return (
        <View style={styles.footerIndicatorStyle}>
          <ActivityIndicator size="large" color={Constants.SECOND_COLOR} />
          <Text>{'Loading...'}</Text>
        </View>
      );
    } else {
      return null;
    }
  };

  renderAllMoviesListItem = ({item, index}) => {
    return (
      <AllMoviesListItem
        item={item}
        index={index}
        onPress={this.props.onPressExpand}
      />
    );
  };

  render() {
    const {allMoviesData, renderTopScreen} = this.props;
    return (
      <FlatList
        ListHeaderComponent={renderTopScreen}
        ListFooterComponent={this.renderFooterIndicator}
        data={allMoviesData}
        keyExtractor={(item, index) => `${index}`}
        renderItem={this.renderAllMoviesListItem}
        {...this.props.getPaginationProps()}></FlatList>
    );
  }
}

const styles = StyleSheet.create({
  footerIndicatorStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
    paddingBottom: 10,
  },
});
