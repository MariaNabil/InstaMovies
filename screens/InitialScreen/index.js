import React, {Component} from 'react';
import {
  StyleSheet,
  View,
  FlatList,
  Dimensions,
  Text,
  TouchableOpacity,
  ActivityIndicator,
} from 'react-native';

import {GetAllMovies} from '../../Services/MoviesServices';
import AllMoviesListItem from './AllMoviesListItem';
import MoviePopup from './MoviePopup';
import MyMoviesListItem from './MyMoviesListItem';
import Constants from '../../utils/Constants';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import LinearGradient from 'react-native-linear-gradient';
import Toast from 'react-native-simple-toast';
import NetInfo from '@react-native-community/netinfo';

export default class InitialScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      screenWidth: Dimensions.get('window').width,
      showPopup: false,
      pressedMovieIndex: -1,
      isMyMoviePressed: false,
      total_pages: 0,
      page: 1,
      allMoviesData: [],
      loading: false,
      isScrollButtonShown: true,
      currentScroll: 0,
      showReloadButton: false,
      lockFetching: false,
      refreshAllMovies: false,
    };
  }

  fetchMovies = () => {
    console.log('fetchMovies');
    NetInfo.fetch().then(({isConnected}) => {
      if (isConnected) {
        this.setState({
          loading: true,
          showReloadButton: false,
          lockFetching: true,
        });
        const {allMoviesData} = this.state;
        this.cancelFetchData = GetAllMovies(
          this.state.page,
          (res) => {
            console.log('componentDidMount res', res, res.data.results);
            this.setState({
              allMoviesData: [...this.state.allMoviesData, ...res.data.results],
              total_pages: res.data.total_pages,
              loading: false,
              lockFetching: false,
            });
          },
          (error) => {
            this.setState({
              loading: false,
              lockFetching: false,
            });
          },
        );
      } else {
        Toast.show(
          'You Are Offline , Please Check Your Internet Connection',
          Toast.LONG,
        );
        this.setState({showReloadButton: true});
      }
    });
  };

  componentDidMount() {
    this.setState({showReloadButton: false});
    this.fetchMovies();
    Dimensions.addEventListener('change', () => {
      this.setState({
        screenWidth: Dimensions.get('window').width,
        screenHeight: Dimensions.get('window').height,
      });
    });
  }

  fetchNextPage = () => {
    this.setState({page: this.state.page + 1});
    console.log('fetchNextPage', this.state.page);
    this.fetchMovies();
  };

  componentWillUnmount() {
    this.cancelFetchData && this.cancelFetchData();
  }

  getPaginationProps = () => {
    return {
      onEndReachedThreshold: 1,
      onEndReached: () => {
        if (
          this.state.page != this.state.total_pages &&
          !this.state.lockFetching
        ) {
          this.fetchNextPage();
        }
      },
    };
  };

  onPressExpand = (index) => {
    console.log('index', index);
    this.setState({
      showPopup: true,
      pressedMovieIndex: index,
      isMyMoviePressed: false,
    });
  };

  onPressMyMovie = (index) => {
    this.setState({
      showPopup: true,
      pressedMovieIndex: index,
      isMyMoviePressed: true,
    });
  };

  renderAllMoviesListItem = ({item, index}) => {
    return (
      <AllMoviesListItem
        item={item}
        index={index}
        onPress={this.onPressExpand}
      />
    );
  };

  renderFooterIndicator = () => {
    if (this.state.loading) {
      return (
        <View
          style={{
            justifyContent: 'center',
            alignItems: 'center',
            alignSelf: 'center',
            paddingBottom: 10,
          }}>
          <ActivityIndicator size="large" color={Constants.SECOND_COLOR} />
          <Text>{'Loading...'}</Text>
        </View>
      );
    } else {
      return null;
    }
  };

  renderAllMoviesList = () => {
    const {allMoviesData} = this.state;
    return (
      <FlatList
        ListHeaderComponent={this.renderFlatListHeaderComponent}
        ListFooterComponent={this.renderFooterIndicator}
        data={allMoviesData}
        keyExtractor={(item, index) => `${index}`}
        renderItem={this.renderAllMoviesListItem}
        refreshing={this.state.refreshAllMovies}
        {...this.getPaginationProps()}></FlatList>
    );
  };

  renderMyMoviesListItem = ({item, index}) => {
    return (
      <MyMoviesListItem
        item={item}
        index={index}
        onPress={this.onPressMyMovie}
      />
    );
  };

  renderHeader = (title) => {
    console.log('renderHeader');
    return (
      <View
        style={{
          marginHorizontal: 10,
          alignSelf: 'stretch',
          flex: 1,
        }}>
        <Text
          style={{
            paddingTop: 10,
            fontSize: 20,
            fontWeight: 'bold',
          }}>
          {title}
        </Text>
      </View>
    );
  };

  handleScroll = (event) => {
    this.setState({currentScroll: event.nativeEvent.contentOffset.x});

    console.log(
      this.state.MyMoviesListWidth,
      event.nativeEvent.contentOffset.x,
    );
  };

  incrementScroll = (increment_by_value) => {
    this.setState({
      currentScroll: this.state.currentScroll + increment_by_value,
    });
    this.listRef.scrollToOffset({offset: +increment_by_value});
  };

  onPressScroll = () => {
    this.incrementScroll(this.state.screenWidth / 2);
  };

  hideScrollButton = () => {
    this.setState({
      isScrollButtonShown: false,
    });
  };

  renderScrollButton = () => {
    if (global.MyMovies?.length > 2 && this.state.currentScroll == 0) {
      const width = 30;

      return (
        <TouchableOpacity
          onPress={this.onPressScroll}
          style={{
            position: 'absolute',
            right: 0,
            backgroundColor: 'transparent',
            top: 0,
            width,
            height: this.state.MyMoviesListHeight,
          }}>
          <LinearGradient
            start={{x: 0.0, y: 0.0}}
            end={{x: 1.0, y: 0.0}}
            colors={[
              `rgba(0, 0, 0, 0.02)`,
              `rgba(0, 0, 0, 0.2)`,
              `rgba(0, 0, 0, 0.5)`,
            ]}
            style={{
              width,
              height: this.state.MyMoviesListHeight,
              justifyContent: 'center',
              alignItems: 'center',
              borderTopLeftRadius: 20,
              borderBottomLeftRadius: 20,
            }}>
            <MaterialIcons
              name={`arrow-forward-ios`}
              size={22}
              color={'white'}
            />
          </LinearGradient>
        </TouchableOpacity>
      );
    }
  };
  renderEmptyList = () => {
    return (
      <TouchableOpacity
        style={{
          width: this.state.screenWidth - 20,
          flex: 1,
          alignItems: 'stretch',
          justifyContent: 'center',
          alignSelf: 'center',
          marginTop: 10,
          marginHorizontal: 5,
        }}
        onPress={() => {
          this.props.navigation.navigate('Add Movie', {
            onSave: this.onReRender,
          });
        }}>
        <View
          style={{
            borderColor: 'black',
            borderWidth: 1,
            borderRadius: 15,
            alignItems: 'center',
            justifyContent: 'center',
          }}>
          <Text
            style={{
              paddingVertical: 5,
              textAlign: 'center',
            }}>
            {'Add Movies'}
          </Text>
        </View>
      </TouchableOpacity>
    );
  };

  renderMyMoviesList = () => {
    const {allMoviesData} = this.state;

    return (
      <FlatList
        onLayout={(e) => {
          this.setState({
            MyMoviesListHeight: e.nativeEvent.layout.height,
            MyMoviesListWidth: e.nativeEvent.layout.width,
          });
        }}
        horizontal={true}
        style={{
          marginHorizontal: 5,
          marginBottom: 10,
        }}
        data={global.MyMovies}
        keyExtractor={({id}) => `${id}`}
        renderItem={({item, index}) =>
          this.renderMyMoviesListItem({item, index})
        }
        ListEmptyComponent={this.renderEmptyList}
        ref={(ref) => (this.listRef = ref)}
        onScroll={this.handleScroll}></FlatList>
    );
  };

  renderMoviePopup = () => {
    const {
      showPopup,
      allMoviesData,
      isMyMoviePressed,
      pressedMovieIndex,
    } = this.state;
    if (showPopup) {
      return (
        <MoviePopup
          allMoviesData={isMyMoviePressed ? global.MyMovies : allMoviesData}
          pressedMovieIndex={pressedMovieIndex}
          showPopup={showPopup}
          onClosePopup={this.onClosePopup}
          isMyMoviePressed={isMyMoviePressed}></MoviePopup>
      );
    } else return null;
  };

  onClosePopup = () => {
    this.setState({showPopup: false});
  };

  renderAddIcon = () => {
    return (
      <TouchableOpacity
        style={{padding: 10}}
        onPress={() => {
          this.props.navigation.navigate('Add Movie', {
            onSave: this.onReRender,
          });
        }}>
        <Ionicons name="add-circle-outline" color="#444444" size={25} />
      </TouchableOpacity>
    );
  };

  renderReloadButton = () => {
    return (
      <TouchableOpacity
        style={{
          flex: 1,
          alignItems: 'stretch',
          justifyContent: 'center',
          alignSelf: 'stretch',
          marginTop: 10,
          marginHorizontal: 10,
        }}
        onPress={() => {
          this.setState({page: 1});
          this.fetchMovies();
        }}>
        <View
          style={{
            borderColor: 'black',
            borderWidth: 1,
            borderRadius: 15,
          }}>
          <Text
            style={{
              paddingVertical: 5,
              textAlign: 'center',
            }}>
            {'Reload'}
          </Text>
        </View>
      </TouchableOpacity>
    );
  };

  onReRender = (refresh) => {
    console.log('onrerender');
    this.setState({
      refreshAllMovies: !this.state.refreshAllMovies,
    });
  };

  renderFlatListHeaderComponent = () => {
    return (
      <View>
        <View style={{flexDirection: 'row', justifyContent: 'space-between'}}>
          {this.renderHeader('My Movies')}
          {global.MyMovies && global.MyMovies.length != 0
            ? this.renderAddIcon()
            : null}
        </View>
        <View>
          {this.renderMyMoviesList()}
          {this.renderScrollButton()}
        </View>
        {this.renderHeader('All Movies')}
        {this.state.showReloadButton ? this.renderReloadButton() : null}
      </View>
    );
  };

  render() {
    console.log('MyMovies', this.state.page);

    return (
      <View style={styles.container}>
        {this.renderAllMoviesList()}
        {this.renderMoviePopup()}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
});
