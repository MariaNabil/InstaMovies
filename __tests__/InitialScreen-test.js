/**
 * @format
 */

import 'react-native';
import React from 'react';
import InitialScreen from '../screens/InitialScreen/index';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('InitialScreen Platform', () => {
  const snap = renderer.create(<InitialScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});
