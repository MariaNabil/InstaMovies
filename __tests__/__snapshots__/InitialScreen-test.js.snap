// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialScreen Platform 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListFooterComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View>
          <View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignSelf": "stretch",
                    "flex": 1,
                    "marginHorizontal": 10,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontSize": 20,
                      "fontWeight": "bold",
                      "paddingTop": 10,
                    }
                  }
                >
                  My Movies
                </Text>
              </View>
            </View>
            <View>
              <RCTScrollView
                ListEmptyComponent={[Function]}
                ListFooterComponent={[Function]}
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={true}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                style={
                  Object {
                    "marginBottom": 10,
                    "marginHorizontal": 5,
                    "marginTop": 5,
                  }
                }
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "stretch",
                        "alignSelf": "center",
                        "flex": 1,
                        "justifyContent": "center",
                        "marginHorizontal": 5,
                        "marginTop": 10,
                        "opacity": 1,
                        "width": 730,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "borderColor": "black",
                          "borderRadius": 15,
                          "borderWidth": 1,
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "paddingVertical": 5,
                            "textAlign": "center",
                          }
                        }
                      >
                        Add Movies
                      </Text>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                  >
                    <View />
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
          <View
            style={
              Object {
                "alignSelf": "stretch",
                "flex": 1,
                "marginHorizontal": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "paddingTop": 10,
                }
              }
            >
              All Movies
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
      />
    </View>
  </RCTScrollView>
</View>
`;
